src/toast-provider.tsx

```diff
@@ -31,45 +31,54 @@ interface state {
     toasts: IToast[];
 }
 
-// @ts-ignore
-export const ToastContext = React.createContext<IToastContext>();
+export const ToastContext = React.createContext<IToastContext>({
+    add: () => {
+        // Default implementation does nothing
+    },
+});
 
 export class ToastProvider extends React.Component<props, state> {
-    state = {toasts: []};
+    // The 'refs' property is required to satisfy the React Component type definition
+    // starting from @types/react v18+. Although it's unused, omitting it causes a TypeScript
+    // error (TS2786) due to stricter class component constructor signatures.
+    // See: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/58632
+    refs: any;
+
+    state: {toasts: IToast[]} = {toasts: []};
 
     public static defaultProps = {
         position: 'top-right',
         maxItems: 8,
         defaultTimer: 4000,
     };
 
-    addToast = (toast: IdlessToast): void => {
+    addToast = (toast: IdlessToast | IToast): void => {
         if (this.state.toasts.length >= this.props.maxItems) {
             this.removeToastByIndex(0, this.state.toasts.length - this.props.maxItems + 1);
         }
 
+        const hasId = 'id' in toast;
         this.setState({
-            toasts: [{...toast, id: new Date().getTime()}, ...this.state.toasts],
+            toasts: [{...toast, id: hasId ? (toast as IToast).id : new Date().getTime()}, ...this.state.toasts],
```

Comment:
Dan kan dit ook zonder de `as`

```suggestion
            toasts: [{...toast, id: isIToast(toast) ? toast.id : new Date().getTime()}, ...this.state.toasts],
```